### PR TITLE
Alphapolis : update getting pages

### LINKED ISSUE #5209: [Alphapolis] Connector not working
**Did you read [the troubleshooting guide](https://hakuneko.download/docs/troubleshoot/)**
Yes

**Is the website of the connector working properly / are you able to see the manga within your browser**
Yes

**Describe the bug**
Can't download the chapters

**To Reproduce**
1. alphapolis.co.jp (https://www.alphapolis.co.jp/manga/official/735000454).
2.  不遇スキルの錬金術師、辺境を開拓する.
4. All of the chapters can't be downloaded.
5. TypeError: Cannot read property 'map' of null
    at <anonymous>:2:2616
    at new Promise (<anonymous>)
    at <anonymous>:2:2475
    at WebFrame../lib/renderer/api/web-frame.ts.WebFrame.<computed> [as executeJavaScript] (web-frame.ts:64)
    at web-frame-init.ts:19
    at EventEmitter.<anonymous> (ipc-renderer-internal-utils.ts:10)
    at EventEmitter.emit (events.js:210)
    at Object.onMessage (init.ts:51) EventTarget {manga: EventTarget, id: "/manga/official/735000454/6580", title: "第10回『大魔法使い様、爆誕？』", file: {…}, language: undefined, …}
_getPageList @ Connector.mjs:618
async function (async)
_getPageList @ Connector.mjs:615
(anonymous) @ Chapter.mjs:111
Promise.then (async)
getPages @ Chapter.mjs:109
downloadPages @ DownloadJob.mjs:72
processQueue @ DownloadManager.mjs:49 

**Screenshots**
image
image

**HakuNeko (please complete the following information):**
 - HakuNeko Version 6.1.7 Portable

### PR DESCRIPTION
**Pull request by MikeZeDev**
Fixes https://github.com/manga-download/hakuneko/issues/5209

### CODE REVIEW

File: src/web/mjs/connectors/Alphapolis.mjs [hunk 1]
```diff
@@ -43,11 +43,20 @@ export default class Alphapolis extends Connector {
...
-        `;
         const request = new Request(new URL(chapter.id, this.url), this.requestOptions);
-        return await Engine.Request.fetchUI(request, script);
+        const data = await this.fetchDOM(request, 'viewer-manga-horizontal');
+        try {
+            const pages = JSON.parse(data[0].getAttribute('v-bind:pages'));
+            let pagesList = [];
+            pages.forEach(page => {
```
Comment (Sheepux) [inline src/web/mjs/connectors/Alphapolis.mjs]: isn't this for loop the same as an array filter ? you could merge both filter in the same one

Comment (MikeZeDev) [reply src/web/mjs/connectors/Alphapolis.mjs]: Solved

File: src/web/mjs/connectors/Alphapolis.mjs [hunk 2]
```diff
@@ -43,11 +43,14 @@ export default class Alphapolis extends Connector {
...
-        `;
         const request = new Request(new URL(chapter.id, this.url), this.requestOptions);
-        return await Engine.Request.fetchUI(request, script);
+        const data = await this.fetchDOM(request, 'viewer-manga-horizontal');
+        try {
+            const pages = JSON.parse(data[0].getAttribute('v-bind:pages'));
+            return pages.filter(element => typeof element != 'object' && !element.match('white_page'));
```
Comment (Sheepux) [inline src/web/mjs/connectors/Alphapolis.mjs:50]: any exemple that explains the use of match ?

Comment (MikeZeDev) [reply src/web/mjs/connectors/Alphapolis.mjs:50]: The json gives us the pictures , but there are junk inside. Some elements are object we dont need, and there is another element that is a regular picture url but just a 'blank page" added at the end of each manga 

image

**Review by Sheepux — APPROVED** [review]

**Pull request merged**